python/cudf/tests/test_multiindex.py

```diff
@@ -0,0 +1,379 @@
+Test related to MultiIndex
+"""
+import pytest
+
+import cudf
+import numpy as np
+import pandas as pd
+
+from cudf.tests.utils import assert_eq
+
+
+def test_multiindex_levels_codes_validation():
+    levels = [['a', 'b'], ['c', 'd']]
+    # Codes not a sequence of sequences
+    with pytest.raises(TypeError):
+        pd.MultiIndex(levels, [0, 1])
+    with pytest.raises(TypeError):
+        cudf.MultiIndex(levels, [0, 1])
+    # Codes don't match levels
+    with pytest.raises(ValueError):
+        pd.MultiIndex(levels, [[0], [1], [1]])
+    with pytest.raises(ValueError):
+        cudf.MultiIndex(levels, [[0], [1], [1]])
+    # Largest code greater than number of levels
+    with pytest.raises(ValueError):
+        pd.MultiIndex(levels, [[0, 1], [0, 2]])
+    with pytest.raises(ValueError):
+        cudf.MultiIndex(levels, [[0, 1], [0, 2]])
+    # Unequal code lengths
+    with pytest.raises(ValueError):
+        pd.MultiIndex(levels, [[0, 1], [0]])
+    with pytest.raises(ValueError):
+        cudf.MultiIndex(levels, [[0, 1], [0]])
+    # Didn't pass levels and codes
+    with pytest.raises(TypeError):
+        pd.MultiIndex()
+    with pytest.raises(TypeError):
+        cudf.MultiIndex()
+    # Didn't pass non zero levels and codes
+    with pytest.raises(ValueError):
+        pd.MultiIndex([], [])
+    with pytest.raises(ValueError):
+        cudf.MultiIndex([], [])
+
+
+def test_multiindex_construction():
+    levels = [['a', 'b'], ['c', 'd']]
+    codes = [[0, 1], [1, 0]]
+    pmi = pd.MultiIndex(levels, codes)
+    mi = cudf.MultiIndex(levels, codes)
+    assert_eq(pmi, mi)
+    pmi = pd.MultiIndex(levels, codes)
+    mi = cudf.MultiIndex(levels=levels, codes=codes)
+    assert_eq(pmi, mi)
+
+
+def test_multiindex_types():
+    codes = [[0, 1], [1, 0]]
+    levels = [[0, 1], [2, 3]]
+    pmi = pd.MultiIndex(levels, codes)
+    mi = cudf.MultiIndex(levels, codes)
+    assert_eq(pmi, mi)
+    levels = [[1.2, 2.1], [1.3, 3.1]]
+    pmi = pd.MultiIndex(levels, codes)
+    mi = cudf.MultiIndex(levels, codes)
+    assert_eq(pmi, mi)
+    levels = [['a', 'b'], ['c', 'd']]
+    pmi = pd.MultiIndex(levels, codes)
+    mi = cudf.MultiIndex(levels, codes)
+    assert_eq(pmi, mi)
+
+
+def test_multiindex_df_assignment():
+    pdf = pd.DataFrame({'x': [1, 2, 3]})
+    gdf = cudf.from_pandas(pdf)
+    pdf.index = pd.MultiIndex([['a', 'b'], ['c', 'd']],
+                              [[0, 1, 0], [1, 0, 1]])
+    gdf.index = cudf.MultiIndex(levels=[['a', 'b'], ['c', 'd']],
+                                codes=[[0, 1, 0], [1, 0, 1]])
+    assert_eq(pdf, gdf)
+
+
+def test_multiindex_series_assignment():
+    ps = pd.Series([1, 2, 3])
+    gs = cudf.from_pandas(ps)
+    ps.index = pd.MultiIndex([['a', 'b'], ['c', 'd']],
+                             [[0, 1, 0], [1, 0, 1]])
+    gs.index = cudf.MultiIndex(levels=[['a', 'b'], ['c', 'd']],
+                               codes=[[0, 1, 0], [1, 0, 1]])
+    assert_eq(ps, gs)
+
+
+def test_string_index():
```

Comment:
This should go into `test_string.py` instead

Reply:
👍